### PR TITLE
User cannot checkout without logging in

### PR DESCRIPTION
**Pull request by dtinianow**

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -4,8 +4,9 @@ def create
     user = User.find_by(username: params[:session][:username])
     if user && user.authenticate(params[:session][:password])
       session[:user_id] = user.id
-      flash[:success] = ""
-      redirect_to dashboard_path
+      flash[:success] = "Successfully logged in as #{params[:session][:username]}"
```
Comment (Yoyo2Code) [inline app/controllers/sessions_controller.rb]: Let's refactor that later into "current_user.username" later once that method has been created. That way we know that there is a user logged in and then we will know for certain that the user is in the database.

**Comment by Yoyo2Code** [comment]
Nice job including the bootstrap already!

**Comment by dtinianow** [comment]
Comments noted.  You will see I also added in both the sessions and user controller create methods, a conditional to redirect the page depending on whether or not the cart is empty.  When the cart is empty, redirect to the user dashboard.  When the cart is not empty, redirect to the cart.

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -4,8 +4,9 @@ def create
     user = User.find_by(username: params[:session][:username])
     if user && user.authenticate(params[:session][:password])
       session[:user_id] = user.id
-      flash[:success] = ""
-      redirect_to dashboard_path
+      flash[:success] = "Successfully logged in as #{current_user.username}"
+      redirect_to cart_path if @cart.contents.any?
```
Comment (Automatic365) [inline app/controllers/sessions_controller.rb]: Good improvement on the base functionality. I don't think the login redirected to the cart previously if it had items in it.

**Pull request merged**